rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java

```diff
@@ -132,6 +132,8 @@ public class TbHttpClient {
             this.webClient = WebClient.builder()
                     .clientConnector(new ReactorClientHttpConnector(httpClient))
                     .defaultHeader(HttpHeaders.CONNECTION, "close") //In previous realization this header was present! (Added for hotfix "Connection reset")
+                    .codecs(configurer -> configurer.defaultCodecs().maxInMemorySize(
+                            (config.getMaxInMemoryBufferSizeInKb() > 0 ? config.getMaxInMemoryBufferSizeInKb() : 256) * 1024))
```

Comment:
A Client can blow out the application by setting and receiving an enormous payload.

For me, the sysadmin should decide the upper limit of the getMaxInMemoryBufferSizeInKb system-wide.

Reply:
Added validation. Hardcoded max size is 25 MB. Can be updated by system parameter.